File: system-test/java/src/main/java/com/optivem/eshop/dsl/driver/port/external/clock/ClockDriver.java
package com.optivem.eshop.dsl.driver.port.external.clock;

import com.optivem.eshop.dsl.driver.port.external.clock.dtos.GetTimeResponse;
import com.optivem.eshop.dsl.driver.port.external.clock.dtos.ReturnsTimeRequest;
import com.optivem.eshop.dsl.driver.port.shared.dtos.ErrorResponse;
import com.optivem.eshop.dsl.common.Result;

public interface ClockDriver extends AutoCloseable {
    Result<Void, ErrorResponse> goToClock();
    Result<GetTimeResponse, ErrorResponse> getTime();
    Result<Void, ErrorResponse> returnsTime(ReturnsTimeRequest request);
}

File: system-test/typescript/src/testkit/driver/port/external/clock/clock-driver.ts
import type { Result } from '../../../../common/result.js';
import type { ClockErrorResponse } from './dtos/errors/ClockErrorResponse.js';
import type { GetTimeResponse } from './dtos/GetTimeResponse.js';
import type { ReturnsTimeRequest } from './dtos/ReturnsTimeRequest.js';

export interface ClockDriver {
  goToClock(): Promise<Result<void, ClockErrorResponse>>;
  getTime(): Promise<Result<GetTimeResponse, ClockErrorResponse>>;
  returnsTime(request: ReturnsTimeRequest): Promise<Result<void, ClockErrorResponse>>;
  close(): Promise<void>;
}
